packages/cosmic-proto/package.json

```diff
@@ -13,50 +13,85 @@
   "exports": {
+    ".": {
+      "types": "./dist/codegen/index.d.ts",
+      "default": "./dist/codegen/index.js"
+    },
     "./package.json": "./package.json",
     "./swingset/msgs.js": {
-      "types": "./dist/agoric/swingset/msgs.d.ts",
-      "default": "./dist/agoric/swingset/msgs.js"
+      "types": "./dist/codegen/agoric/swingset/msgs.d.ts",
+      "default": "./dist/codegen/agoric/swingset/msgs.js"
     },
     "./swingset/query.js": {
-      "types": "./dist/agoric/swingset/query.d.ts",
-      "default": "./dist/agoric/swingset/query.js"
+      "types": "./dist/codegen/agoric/swingset/query.d.ts",
+      "default": "./dist/codegen/agoric/swingset/query.js"
     },
     "./vstorage/query.js": {
-      "types": "./dist/agoric/vstorage/query.d.ts",
-      "default": "./dist/agoric/vstorage/query.js"
-    },
-    "./swingset/swingset.js": {
-      "types": "./dist/agoric/swingset/swingset.d.ts",
-      "default": "./dist/agoric/swingset/swingset.js"
+      "types": "./dist/codegen/agoric/vstorage/query.d.ts",
+      "default": "./dist/codegen/agoric/vstorage/query.js"
     }
   },
+  "main": "dist/index.js",
+  "module": "dist/index.mjs",
+  "typings": "dist/index.d.ts",
+  "directories": {
+    "lib": "src"
+  },
+  "files": [
+    "dist",
+    "!CHANGELOG.md"
+  ],
   "scripts": {
-    "build": "echo Use yarn rebuild to update dist output",
-    "rebuild": "rm -rf gen && mkdir -p gen && yarn generate && tsc --build && yarn prettier -w dist/agoric && rm -rf gen",
-    "generate": "protoc --plugin=node_modules/.bin/protoc-gen-ts_proto --ts_proto_opt='esModuleInterop=true,forceLong=long,useOptionals=messages' --ts_proto_out=./gen --ts_proto_opt=importSuffix=.js ./proto/agoric/swingset/msgs.proto ./proto/agoric/swingset/query.proto ./proto/agoric/vstorage/query.proto -I./proto",
-    "test": "node test/sanity-test.js",
+    "build:cjs": "yarn tsc --project tsconfig.build.json --outDir dist --module commonjs || true",
+    "build:mjs": "yarn tsc --project tsconfig.build.json --outDir mjs --module es2022 --declaration false || true",
+    "clean:mjs": "rimraf mjs",
+    "clean:dist": "rimraf dist",
+    "clean": "npm run clean:mjs && npm run clean:dist",
+    "build:rename": "publish-scripts --cmd rename --srcDir mjs --outDir dist --findExt js --replaceExt mjs --no-rmDir",
+    "build": "npm run clean && npm run build:cjs && npm run build:mjs && npm run build:rename",
+    "codegen": "yarn protos-update && node scripts/codegen.cjs && yarn prettier --write src",
+    "prepare": "npm run build",
+    "protos-update": "cp -rf ../../golang/cosmos/third_party/proto . && cp -rf ../../golang/cosmos/proto/agoric proto",
+    "lint-fix": "yarn lint:eslint --fix",
+    "lint": "tsc",
+    "test": "ava",
     "test:xs": "exit 0",
-    "lint": "exit 0",
-    "lint-fix": "exit 0"
+    "test:live": "yarn build && scripts/test-live.mjs"
   },
-  "dependencies": {
-    "protobufjs": "^7.0.0"
+  "publishConfig": {
+    "access": "restricted"
```

Comment:
Reallly?  Shouldn't this be `"access": "public"`?

Reply:
yeah, that's a safe default from Telscope's generator. will revise